BlueDBOnDisk/src/main/java/io/bluedb/disk/segment/Segment.java

```diff
@@ -40,41 +41,104 @@ public boolean contains(BlueKey key) throws BlueDbException {
 		return get(key) != null;
 	}
 
-	public void save(BlueKey key, T value) throws BlueDbException {
-		ArrayList<BlueEntity<T>> entities = null;
-		File file;
-		try (BlueReadLock<Path> readLock = getReadLockFor(key)) {
-			if (readLock != null) {
-				entities = fetch(readLock);
-				file = readLock.getKey().toFile();
-			} else {
-				entities = new ArrayList<>();
-				file = 	getPathFor(key, 1).toFile();
+	public void update(BlueKey newKey, T newValue) throws BlueDbException {
+		long groupingNumber = newKey.getGroupingNumber();
+		modifyChunk(groupingNumber, new Processor<T>() {
+			@Override
+			public void process(BlueObjectInput<BlueEntity<T>> input, BlueObjectOutput<BlueEntity<T>> output) throws BlueDbException {
+				BlueEntity<T> newEntity = new BlueEntity<T>(newKey, newValue);
+				while (input.hasNext()) {
+					BlueEntity<T> iterEntity = input.next();
+					BlueKey iterKey = iterEntity.getKey();
+					if (iterKey.equals(newKey)) {
+						output.write(newEntity);
+						newEntity = null;
+					} else if (newEntity != null && iterKey.getGroupingNumber() > groupingNumber) {

```

Comment:
So if I delete something then try to update it then the object will be resurrected? Or I guess in order to have the newValue that means that we already found the value once and applied the update lambda to it right? Why do we end up needing this branch?

Reply:
removed recreate on update